src/dawn/engine_dawn.cc

```diff
@@ -471,67 +608,253 @@ Result EngineDawn::DoClear(const ClearCommand* command) {
+  renderPipelineDescriptor.primitiveTopology =
+      ::dawn::PrimitiveTopology::TriangleList;
+  renderPipelineDescriptor.sampleCount = 1;
+
+  // Set defaults for the vertex stage descriptor.
+  snprintf(entryPoint,
+           sizeof(render_pipeline.pipeline->GetShaders()[0].GetEntryPoint()),
+           "%s",
+           render_pipeline.pipeline->GetShaders()[0].GetEntryPoint().c_str());
+  cVertexStage.module = render_pipeline.vertex_shader;
+  cVertexStage.entryPoint = entryPoint;
+  renderPipelineDescriptor.vertexStage = &cVertexStage;
+
+  // Set defaults for the fragment stage descriptor.
+  cFragmentStage.module = render_pipeline.fragment_shader;
+  cFragmentStage.entryPoint = entryPoint;
+  renderPipelineDescriptor.fragmentStage = std::move(&cFragmentStage);
+
+  // Set defaults for the input state descriptors.
+  tempInputState.indexFormat = ::dawn::IndexFormat::Uint32;
+  // Fill the default values for vertexInput.
+  vertexInput.inputSlot = 0;
+  vertexInput.stride = 0;
+  vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
+  for (uint32_t i = 0; i < kMaxVertexInputs; ++i) {
+    tempInputs[i] = vertexInput;
+  }
+  tempInputs[0].stride = 4 * sizeof(float);
+  tempInputState.inputs = std::move(&tempInputs[0]);
+  // Fill the default values for vertexAttribute.
+  vertexAttribute.shaderLocation = 0;
+  vertexAttribute.inputSlot = 0;
+  vertexAttribute.offset = 0;
+  vertexAttribute.format = ::dawn::VertexFormat::Float;
+  for (uint32_t i = 0; i < kMaxVertexAttributes; ++i) {
+    tempAttributes[i] = vertexAttribute;
+  }
+  tempAttributes[0].format = ::dawn::VertexFormat::Float4;
+  tempInputState.attributes = &tempAttributes[0];
+  tempInputState.numAttributes = 1;
+  tempInputState.numInputs = 1;
+  renderPipelineDescriptor.inputState = &tempInputState;
+
+  /// Set defaults for the color state descriptors.
+  renderPipelineDescriptor.colorStateCount = 1;
+  blend.operation = ::dawn::BlendOperation::Add;
+  blend.srcFactor = ::dawn::BlendFactor::One;
+  blend.dstFactor = ::dawn::BlendFactor::Zero;
+  colorStateDescriptor.format = fb_format;
+  colorStateDescriptor.alphaBlend = blend;
+  colorStateDescriptor.colorBlend = blend;
+  colorStateDescriptor.colorWriteMask = ::dawn::ColorWriteMask::All;
+  for (uint32_t i = 0; i < kMaxColorAttachments; ++i) {
+    mColorStates[i] = colorStateDescriptor;
+    cColorStates[i] = &mColorStates[i];
+  }
+  cColorStates[0]->format = fb_format;
+  renderPipelineDescriptor.colorStates = &cColorStates[0];
+
+  /// Set defaults for the depth stencil state descriptors.
+  stencilFace.compare = ::dawn::CompareFunction::Always;
+  stencilFace.failOp = ::dawn::StencilOperation::Keep;
+  stencilFace.depthFailOp = ::dawn::StencilOperation::Keep;
+  stencilFace.passOp = ::dawn::StencilOperation::Keep;
+  cDepthStencilState.format = fb_format;
+  cDepthStencilState.depthWriteEnabled = false;
+  cDepthStencilState.depthCompare = ::dawn::CompareFunction::Always;
```

Comment:
I'm not certain this is right, but ... hey as long as the tests pass.  :-)

Reply:
Neither am I :) I've done what Dawn examples do.